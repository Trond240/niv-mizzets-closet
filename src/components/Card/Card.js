import React from 'react';
import './Card.scss';
import { Link } from 'react-router-dom';

export const Card = ({ name, image, id, setSelectedCard, clearSelectedCardInfo }) => {
  return (
    <Link 
      to={`/cards/${id}`} 
      className='card' 
      id={id} 
      onClick={() => {
        setSelectedCard(id)}}
    >
      <img src={image} alt={`${name} card picture`} />
    </Link>
  )
}