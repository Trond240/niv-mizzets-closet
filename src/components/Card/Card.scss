@import '../../styles/mixin.scss';
@import '../../styles/variables.scss';

.card {
  @include box-shadow(0 0 0 0, 0 0 1em 0);
  border-radius: 1em;
  margin: 1em;
  width: 20em;
  &__image {
    border-radius: 1em;
    width: 20em;
  }
}