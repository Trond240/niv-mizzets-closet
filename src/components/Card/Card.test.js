import React from 'react';
import { shallow } from 'enzyme';
import { Card } from './Card.js';

describe('Card', () => {
  let wrapper;
  let passedProps = {setSelectedCard: jest.fn()}

  beforeEach(() => {
    wrapper = shallow(<Card {...passedProps} />)
  })

  it ('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it ('should call the setSelectedCard function when link is clicked', () => {
    wrapper.find('.card').simulate('click');
    expect(passedProps.setSelectedCard).toHaveBeenCalled();
  })
})