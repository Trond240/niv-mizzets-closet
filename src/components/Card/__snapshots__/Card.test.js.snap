// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match a snapshot 1`] = `
<Link
  className="card"
  onClick={[Function]}
  to="/cards/undefined"
>
  <img
    alt="undefined card picture"
    className="card__image"
  />
</Link>
`;
